bot/exts/events/advent_of_code/_cog.py

```diff
@@ -185,15 +186,47 @@ async def join_leaderboard(self, ctx: commands.Context) -> None:
         brief="Get a snapshot of the PyDis private AoC leaderboard",
     )
     @whitelist_override(channels=AOC_WHITELIST_RESTRICTED)
-    async def aoc_leaderboard(self, ctx: commands.Context) -> None:
-        """Get the current top scorers of the Python Discord Leaderboard."""
+    async def aoc_leaderboard(
+            self,
+            ctx: commands.Context,
+            day_and_star: Optional[str] = None,
+            maximum_scorers: Optional[int] = 10
+    ) -> None:
+        """
+        Get the current top scorers of the Python Discord Leaderboard.
+
+        Additionally, you can provide a day-star(e.g. 2-1, as second day first star) to filter your query more.
+        """
+        if maximum_scorers > AocConfig.max_day_and_star_results or maximum_scorers <= 0:
+            raise commands.BadArgument(
+                "Unfortunately, you reached a limit. "
+                f"(The maximum number of results you can query is {AocConfig.max_day_and_star_results})"
+            )
         async with ctx.typing():
             try:
                 leaderboard = await _helpers.fetch_leaderboard()
-            except _helpers.FetchingLeaderboardFailed:
+            except _helpers.FetchingLeaderboardFailedError:
                 await ctx.send(":x: Unable to fetch leaderboard!")
                 return
-
+            if day_and_star:
```

Comment:
Because of how `day_and_star` has more code than the "else", can we flip this? So `if not day_and_star` and move the code outside the if-statement into it and vice versa.

Reply:
Of course, I missed that one! 👍